ops/_tracing/__init__.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The tracing facility of the Operator Framework.
+
+TODO: quick start, usage example.
+"""
+
+from __future__ import annotations
+
+import logging
+
+from ops._tracing import hacks
+
+# FIXME must this hack be run before OTEL packages are imported?
+hacks.remove_stale_otel_sdk_packages()
+
+
+try:
+    from . import export
+except ImportError:
+    export = None
+
+
+def setup_tracing(charm_class_name: str) -> None:
+    """Setup tracing for this "dispatch" of the charm code."""
+    if not export:
+        return
+    export.setup_tracing(charm_class_name)
+
+
+def set_tracing_destination(
+    *,
+    url: str | None,
+    # FIXME: API design choice, decide on CA semantics:
+    # - a local path to a file with CA data
+    # - or the CA data itself?
+    #
+    # Sadly Requests `verify=` kwarg accepts only:
```

Comment:
Not sure how fixed the current choice is, or whether the change from requests to urllib alters things.

I would think that given that the CA probably comes from a secret (via a relation) it would be much more convenient to be the actual data and not a path (and if you do have a path, it's pretty simple to read it first, whereas you incur a penalty if you have it in memory but not a file).

Reply:
Indeed, ca from relation should be PEM content.

IMHO, the only use case for path is using the system CA list which may be stored as a bunch of files in the directory.

Confirming with COS: https://matrix.to/#/!nHXpRkcSNJHlHdUGbQ:ubuntu.com/$A7zFMyAKTDLiGfyilvHcBpL7hiuHAYEXTuSG6d_cUI8?via=ubuntu.com&via=matrix.org